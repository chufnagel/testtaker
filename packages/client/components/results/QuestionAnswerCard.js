import PropTypes from "prop-types";
import Question from "../Question";

// iterate through answers
// highlight correct answer with green text
// if correct !== userAnswer
// highlight incorrect answer with red text
const getClassName = (correct, choice, userAnswer) => {
  let className = "radio-row";
  if (correct === choice) {
    className = "radio-row correct";
  } else if (correct !== userAnswer && userAnswer === choice) {
    className = "radio-row incorrect";
  }
  return className;
};

const QuestionAnswerCard = ({ id, text, correct, userAnswer, answers }) => {
  return (
    <div className="answer-card card">
      <Question id={id} text={text} />
      {userAnswer === correct ? (
        <p className="correct">Great!</p>
      ) : (
        <p className="incorrect">
          You answered {userAnswer}, but the right answer was {correct}
        </p>
      )}
      {answers.map(answer => (
        <Answer
          questionId={answer.questionId}
          choice={answer.choice}
          key={answer.choice}
          text={answer.content}
          checked={correct === answer.choice}
          className={getClassName(correct, answer.choice, userAnswer)}
        />
      ))}
    </div>
  );
};

QuestionAnswerCard.propTypes = {
  id: PropTypes.number,
  text: PropTypes.string,
  correct: PropTypes.string,
  userAnswer: PropTypes.string,
  answers: PropTypes.arrayOf(PropTypes.object),
};

QuestionAnswerCard.defaultProps = {
  id: 1,
  text: "Navita (line 2) refers to",
  correct: "A",
  userAnswer: "B",
  answers: [
    {
      choice: "A",
      content: "Evander",
    },
    {
      choice: "B",
      content: "Latinus",
    },
    {
      choice: "C",
      content: "Priam",
    },
    {
      choice: "D",
      content: "Juturna",
    },
  ],
};

const Answer = ({ questionId, choice, text, checked, correct, className }) => (
  <div
    className={className}
    key={`${questionId}${choice}`}
    aria-checked={checked}
    checked={checked}
    role="radio"
    tabIndex="0"
  >
    <input key={`${questionId}:${choice}`} name="answer" value={text} type="radio" />
    {`${choice} - `} {`${text}`}
    <br />
  </div>
);

export default QuestionAnswerCard;
