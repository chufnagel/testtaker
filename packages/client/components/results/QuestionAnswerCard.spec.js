/* global beforeEach describe, expect, test */
import { shallow } from "enzyme";
import QuestionAnswerCard from "./QuestionAnswerCard";

// QuestionAnswerCard.propTypes = {
//   id: PropTypes.number,
//   text: PropTypes.string,
//   correct: PropTypes.string,
//   userAnswer: PropTypes.string,
//   answers: PropTypes.arrayOf(PropTypes.object),
// };

describe("QuestionAnswerCard", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<QuestionAnswerCard />);
  });
  test("It should render without crashing", () => {
    expect(wrapper.exists()).toEqual(true);
  });
  test("QuestionAnswerCard matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
