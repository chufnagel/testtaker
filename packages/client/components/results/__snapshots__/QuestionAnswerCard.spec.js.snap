// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionAnswerCard QuestionAnswerCard matches snapshot 1`] = `
<div
  className="answer-card card"
>
  <Question
    id={1}
    text="Navita (line 2) refers to"
  />
  <p
    className="incorrect"
  >
    You answered 
    B
    , but the right answer was 
    A
  </p>
  <Answer
    checked={true}
    choice="A"
    className="radio-row correct"
    key="A"
    text="Evander"
  />
  <Answer
    checked={false}
    choice="B"
    className="radio-row incorrect"
    key="B"
    text="Latinus"
  />
  <Answer
    checked={false}
    choice="C"
    className="radio-row"
    key="C"
    text="Priam"
  />
  <Answer
    checked={false}
    choice="D"
    className="radio-row"
    key="D"
    text="Juturna"
  />
</div>
`;
